Lib/fontTools/ufoLib/glifLib.py

```diff
@@ -1240,8 +1244,11 @@ def _readAdvance(glyphObject, advance):
 	_relaxedSetattr(glyphObject, "height", height)
 
 def _readNote(glyphObject, note):
-	lines = note.text.split("\n")
-	note = "\n".join(line.strip() for line in lines if line.strip())
+	if not _noteIsCDATA(note):
+		lines = note.text.split("\n")
+		note = "\n".join(line.strip() for line in lines if line.strip())
+	else:
+		note = note.text
```

Comment:
I don't think the parsing of a note element containing CDATA section is working as you think it is.

In lxml, CDATA sections are _not_ preserved in the text property of an element even if `strip_cdata=False` is used, see https://lxml.de/api.html#cdata.

In fact codecov bot complains that this line 1251 is never hit by your tests.
Try to add some empty new lines or leading/trailing whitespace to your test CDATA section, and you'll see they are being stripped upon parsing. E.g. 

```python
		s = textwrap.dedent("""\
		<?xml version='1.0' encoding='UTF-8'?>
		<glyph name="asterisk" format="2">
		  <note><![CDATA[반갑슴니다 &amp;
			    etc. <!-- [] --> 

		]]></note>
		</glyph>
		""")
```

... pytest will fail with

```
    	readGlyphFromString(s, g, pen, validate=False)
>   	assert g.note == r"반갑슴니다 &amp; etc. <!-- [] -->"
E    AssertionError: assert '반갑슴니다 &amp;\.... <!-- [] -->' == '반갑슴니다 &amp; etc. <!-- [] -->'
E      - 반갑슴니다 &amp; etc. <!-- [] -->
E      + 반갑슴니다 &amp;
E      + etc. <!-- [] -->

Tests/ufoLib/glifLib_test.py:306: AssertionError
```

The only way these opaque CDATA objects manifest themselves to lxml Python API is when you _serialize_ the element (e.g. with etree.tostring()).
Now, you could potentially do that and check if CDATA is present in the string:

```diff
diff --git a/Lib/fontTools/ufoLib/glifLib.py b/Lib/fontTools/ufoLib/glifLib.py
index 0e3809fae..60998929b 100755
--- a/Lib/fontTools/ufoLib/glifLib.py
+++ b/Lib/fontTools/ufoLib/glifLib.py
@@ -14,6 +14,7 @@ from __future__ import annotations

 import logging
 import enum
+import re
 from warnings import warn
 from collections import OrderedDict
 import fs
@@ -847,12 +848,19 @@ def _writeUnicodes(glyphObject, element, validate):
 		hexCode = "%04X" % code
 		etree.SubElement(element, "unicode", dict(hex=hexCode))

+CDATA_RE = re.compile(r"<!\[CDATA\[(.*)\]\]>", flags=re.DOTALL | re.MULTILINE)
+
 def _noteIsCDATA(note):
-	return (note.__class__.__name__, note.__class__.__module__) == ("CDATA", "lxml.etree")
+	# Return True if the note text contains a CDATA section
+	if getattr(note, "text", None) is not None:
+		s = etree.tostring(note, encoding="unicode").strip()
+		if CDATA_RE.search(s):
+			return True
+	return False
```

... but I think it would overkill as it would slow down parsing for the 99% of the case notes where do not contain CDATA. So I'm -1 on that.

I think we should content ourselves of simply allowing to write CDATA unchanged, but for parsing we'd have to live with what ElementTree or lxml give us. And if both, CDATA are parsed and resolved to plain text strings leaving no visible trace behind.